sale_measure/measure.py

```diff
@@ -0,0 +1,104 @@
+""" The goal of this program is make a measure with different
+    forms for example trousers,skirt,.... and for each form there are
+    differnt fields, so we hide or show fields as a function of forms  """
+
+from openerp import fields, models, api, _
+from collections import defaultdict
+from openerp.exceptions import Warning
+from openerp.osv import orm
+from lxml import etree
+
+
+class MeasureMeasure(models.Model):
+    """Measure Class"""
+    _name = 'measure.measure'
+    _description = "Measure for each partner"
+
+    product_id = fields.Many2one(
+        'product.product',
+        string="Product",
+        required=True)
+    partner_id = fields.Many2one(
+        'res.partner',
+        string="partner",
+        required=True)
+    measure_form_type = fields.Selection(
+        related='product_id.measure_form_type',
+        readonly=True)
+    sale_line_ids = fields.One2many(
+        'sale.order.line',
+        'measure_id',
+        string="Measure Ligne")
+    quantite = fields.Float('Quantity')
+
+    @classmethod
+    def _get_form(cls):
+        return {}
+
+    @api.model
+    def get_form(self):
+        """ inherited function so we have a same function in custom
+             this function return dictionary with all forms attribute """
+        return self._get_form()
```

Comment:
We need a better solution for returning the form with the field name instead of the field key